### PR TITLE
Issue #40 - document use of Functions - Command Query Separation

### PR DESCRIPTION
**Pull request by wintondeshong**
Fixes issue #40 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [ ] *n/a* Code builds cleanly (no _additional_ warnings or errors)
- [ ] *n/a* Manually tested
- [ ] *n/a* Automated tests are passing
- [ ] *n/a* No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [ ] *n/a* Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Review by Stefanie899 — APPROVED** [review]

File: clean-code/FUNCTIONS.md [hunk 1]
```diff
@@ -129,8 +129,44 @@ function assertExpectedEqualsActual(expected: any, actual: any) { ... }
...
+```
+
+_NOT_ both...
+
+```TypeScript
+/**
+ * Cache key value pair to local storage as strings.
+ * Returns value of key if successful or undefined if unsuccessful
+ */
+function set<T>(key: string, value: T): void {
```
Comment (jstevenson) [inline clean-code/FUNCTIONS.md:165]: This is missing the bool return type

**Review by jstevenson — APPROVED** [review]

File: clean-code/FUNCTIONS.md [hunk 2]
```diff
@@ -129,8 +129,44 @@ function assertExpectedEqualsActual(expected: any, actual: any) { ... }
...
+    localStorage.setItem(key, JSON.stringify(value));
+}
+```
+
+Or **Answer** something...
+
+```TypeScript
+// role-record.ts
```
Comment (brandongregoryscott) [inline clean-code/FUNCTIONS.md:148]: Nitpick - I think the comments here are off (references to `role-record`, `RoleRecord`, `type`

File: clean-code/FUNCTIONS.md [hunk 3]
```diff
@@ -129,8 +129,44 @@ function assertExpectedEqualsActual(expected: any, actual: any) { ... }
...
+    return localStorage.getItem(key) != null;
+}
+```
+
+_NOT_ both...
+
+```TypeScript
+/**
+ * Cache key value pair to local storage as strings.
+ * Returns value of key if successful or undefined if unsuccessful
```
Comment (brandongregoryscott) [inline clean-code/FUNCTIONS.md:163]: Believe this would actually be something like 'Returns true if key successfully set, or false if key not set'

**Review by brandongregoryscott — APPROVED** [review]

File: clean-code/FUNCTIONS.md (hunk 1 above)
Comment (wintondeshong) [reply clean-code/FUNCTIONS.md:165]: good catch

File: clean-code/FUNCTIONS.md (hunk 2 above)
Comment (wintondeshong) [reply clean-code/FUNCTIONS.md:148]: Doh, yeah initially had an example from role record, changed my mind and never revisited that. Good catch!

File: clean-code/FUNCTIONS.md (hunk 3 above)
Comment (wintondeshong) [reply clean-code/FUNCTIONS.md:163]: Same as above - missed reworking the comments after changing them. Good catch!

**Pull request merged**